test/Solver/range1.ll

```diff
@@ -0,0 +1,16 @@
+; REQUIRES: solver
+
+; RUN: %llvm-as -o %t %s
+; RUN: %souper %solver -check %t 2> %t2 || true
+; RUN: %FileCheck %s < %t2
+
+define i1 @cmp_with_range(i8*, i8*) {
+  %v1 = load i8, i8* %0, !range !0
+  %v2 = load i8, i8* %1, !range !1
+  ; CHECK: cand %2 0:i1
+  %out = icmp eq i8 %v1, %v2
+  ret i1 %out
+}
+
+!0 = !{i8 5, i8 7}

```

Comment:
please add at least one LLVM IR test where the range information provided is too weak to make an optimization happen

Reply:
Yes, the last thing to be added is this.